### DISCUSSION TITLE
Shift Alternative Consult

### DISCUSSION THREAD

**Discussion by gabrieltagleh** [pydata/xarray#5856] [Q&A]
Hi Everyone! I use xarray for financial planning models.

 I´m having trouble to calculate values in arrays, where one value depends on previous period value.

Shift() function does not attends when the value of an array is calculated in function of a previous value from the same array.

I would be really thankfull if you or someone can take a I look to this case.

```python
import xarray as xr
import pandas as pd

region_ind = pd.Index(["North","South","West"])
product_ind = pd.Index(["Orange","Apple","Banana"])

region = xr.Coordinate( 'Regions', region_ind)
product = xr.Coordinate( 'Products', product_ind)

fruits = xr.DataArray([[15.0,20.0,25.0], [6.0,5.0,0.0], [0.0,0.0,25.0]],[region, product])

periods_ind = pd.Index(["Period 1", "Period 2", "Period 3"])
periods =  xr.Coordinate( 'Periods', periods_ind)

prices =  xr.DataArray([[1,2,3],[2,4,6],[3,6,9]],[product, periods])

revenue = fruits*prices
```

**I need to calculate the adjusted revenue of future periods as a function of a previous period value. Where I have the 99, I would like to complete with "2 Previous Period Value * X%".**

```python
flag_adjusted_revenue = xr.DataArray([1,1,0],[periods])
revenue_by_periods = xr.where(flag_adjusted_revenue==1, revenue, 99)

revenue_by_periods
```

**Comment by dcherian** [comment]
Hello  that example code (first block) does not run. So it is hard for  us to help you. In general, you should not have  to use `pd.Index` or `xr.Coordinate`. See https://xarray.pydata.org/en/stable/user-guide/data-structures.html#creating-a-dataarray for how to create  a dataarray directly.

**Reply by gabrieltagleh** [reply]
Hello Deepak, thank you very much for your answer. I repost the code again with the changes:

```python
import xarray as xr
import pandas as pd

regions = ["North","South","West"]
products = ["Orange","Apple","Banana"]
q_data = [[15.0,20.0,25.0], [6.0,5.0,0.0], [0.0,0.0,25.0]]

quantities = xr.DataArray(q_data, coords=[regions, products], dims=['Regions', 'Products'])

periods = ["Period 1", "Period 2", "Period 3"]
p_data = [[1,2,3],[2,4,6],[3,6,9]]

prices =  xr.DataArray(p_data, coords=[products, periods], dims=['Products', 'Periods'])

revenue = quantities * prices
```

**I need to calculate the adjusted revenue of future periods as a function of a previous period value. Where I have the 99, I would like to complete with "2 Previous Period Value * X%".**

```python
flag_adjusted_revenue = xr.DataArray([1,1,0], coords=[periods], dims=['Periods'])
revenue_by_periods = xr.where(flag_adjusted_revenue == 1, revenue, 99.)

revenue_by_periods
```

**Thanks!!**

**Reply by krisjune** [reply]
Have you tried the below?
`revenue_by_periods = revenue`
`revenue_by_periods = revenue_by_periods.where(flag_adjusted_revenue == 1,revenue.shift(Periods=2)*1.15)`
